extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -291,7 +291,8 @@ export class PublishDatabaseDialog {
 			// We only use the dialog in ADS context currently so safe to cast to the mssql DeploymentOptions here
 			this.deploymentOptions = await utils.getDefaultPublishDeploymentOptions(this.project) as DeploymentOptions;
 		}
-
+		// convert Map table into JSON object to send to STS
```

Comment:
1. Is this even valid? You're assigning an object type to optionsMapTable here but the type is a Map so I'm surprised it let you do that. And regardless you shouldn't be doing that since now the type is incorrect so anyone trying to use that isn't going to get the correct value.

Is there a reason you're using a Map type in the first place? I'd suggest just making a custom type 

`export type DeploymentOptionsMap = { [key: string]: DacDeployOptionPropertyBoolean };`

That way it's already just a JSON object (which are essentially maps already)

Reply:
No reason to use Map, using it because it's better to use Map instead of its type Record<>. yeah, like the custom type, will make changes with this! Thanks.

Reply:
@Charles-Gagnon , modified the code accordingly! Thanks

Reply:
Doesn't this also need to be updated in STS to match? It would also be good to change the name from OptionsMapTable to DeploymentOptionsMap, as suggested above - it's confusing having both "map" and "table" in the name.